src/drivers/bitbucket_cloud.js

```diff
@@ -22,39 +22,66 @@ class BitBucketCloud {
-    const commitOutput = await this.request({
-      endpoint: commitEndpoint,
-      method: 'POST',
-      body: commitBody
-    });
+
+    const existingCommmit = (
+      await this.paginatedRequest({ endpoint: commitEndpoint, method: 'GET' })
+    )
+      .filter(
+        (comment) =>
+          comment.content.raw && comment.content.raw.endsWith(watermark)
+      )
+      .sort((first, second) => first.id < second.id)
+      .pop();
+
+    const commitOutput = (
+      await this.request({
+        endpoint:
+          commitEndpoint +
+          (update && existingCommmit ? existingCommmit.id : ''),
+        method: update && existingCommmit ? 'PUT' : 'POST',
+        body: JSON.stringify({ content: { raw: report } })
+      })
+    ).links.html.href;
 
     // Check for a corresponding PR. If it exists, also put the comment there.
-    const getPrEndpt = `/repositories/${projectPath}/commit/${commitSha}/pullrequests`;
-    const { values: prs } = await this.request({ endpoint: getPrEndpt });
+    let prs;
+    try {
+      const getPrEndpoint = `/repositories/${projectPath}/commit/${commitSha}/pullrequests`;
+      prs = await this.paginatedRequest({ endpoint: getPrEndpoint });
+    } catch (err) {
+      if (err.message === 'Not Found Resource not found')
+        console.warn(
+          "Can't create a pull request comment: the Pull Request Commit Links application has not been installed."
+        );
+      else throw err;
+    }
 
     if (prs && prs.length) {
       for (const pr of prs) {
-        try {
-          // Append a watermark to the report with a link to the commit
-          const commitLink = commitSha.substr(0, 7);
-          const longReport = `${commitLink}   \n${report}`;
-          const prBody = JSON.stringify({ content: { raw: longReport } });
-
-          // Write a comment on the PR
-          const prEndpoint = `/repositories/${projectPath}/pullrequests/${pr.id}/comments`;
-          await this.request({
-            endpoint: prEndpoint,
-            method: 'POST',
-            body: prBody
-          });
-        } catch (err) {
-          console.debug(err.message);
-        }
+        // Append a watermark to the report with a link to the commit
+        const commitLink = commitSha.substr(0, 7);
+        const longReport = `${commitLink}   \n${report}`;
+        const prBody = JSON.stringify({ content: { raw: longReport } });
+
+        // Write a comment on the PR
+        const prEndpoint = `/repositories/${projectPath}/pullrequests/${pr.id}/comments/`;
+        const existingPr = (
+          await this.paginatedRequest({ endpoint: prEndpoint, method: 'GET' })
+        )
+          .filter(
+            (comment) =>
+              comment.content.raw && comment.content.raw.endsWith(watermark)
+          )
+          .sort((first, second) => first.id < second.id)
+          .pop();
+        await this.request({
+          endpoint: prEndpoint + (update && existingPr ? existingPr.id : ''),
```

Comment:
can we assing first the vars and then:

```js
await this.request({ endpoint, method, body })
```

Reply:
We _can_, but having a new variable named `endpoint` in a context where we are going to need both might be counterintuitive. Unless you have a strong preference for the suggested approach, I would prefer to keep it as is.

Reply:
same as before, is a matter of style

Reply:
🤷🏼  As usual, feel free to update it to fit the style.